### PR TITLE
support for redis as a transport

### LINKED ISSUE #556: Doesn't create a transport.json file
It would seem that the latest version of the sensu requires a transport.json file, configured for either redis or rabbitmq. I suppose in the past it defaulted to rabbitmq and this file wasn't needed, but now it does. 

`{
  "transport": {
    "name": "redis",
    "reconnect_on_error": true
  }
}`

### PR DESCRIPTION
**Pull request by RiRa12621**
this fixes #556

this work is heavily based on the work of @raffraffraff

this should be supported ASAP

### CODE REVIEW

**Comment by gutmensch** [comment]
LGTM

**Comment by lgarciaaco** [comment]
This is awesome!!

File: manifests/init.pp [hunk 1]
```diff
@@ -196,6 +196,11 @@
 #   Boolean.  Reconnect to Redis in the event of a connection failure
 #   Default: true
 #
+# [*transport_types*]
+#   String. Transport type to be used by Sense
```
Comment (cryptk) [inline manifests/init.pp]: This should probably say Sensu rather than Sense.

Comment (RiRa12621) [inline manifests/init.pp]: maybe

Comment (RiRa12621) [inline manifests/init.pp]: fixed

**Comment by cwjohnston** [comment]
Hi @RiRa12621, thanks for the pull request. This implementation seems reasonable, but I'd like to point out a few things for those following this PR.

* There are unresolved issues with the Redis transport which make it less than suitable for production, i.e. [duplication of check requests](https://github.com/sensu/sensu-transport/issues/31). As a result we recommend the Redis transport be used only in testing and development scenarios.

* Sensu 0.19 is now almost two years old by now, and no longer available from the official Sensu package repositories. Anyone using a version prior to 0.23 is susceptible to Heartbleed, among other vulnerabilities.

* The next release of this module will include other changes already merged to master which are thought to be incompatible with Sensu versions prior 0.27.

With regard to the last two points, I think its reasonable to remove logic guarding redis transport on Sensu < 0.19 as that code will most likely never be exercised.

**Comment by RiRa12621** [comment]
Hey @cwjohnston,

- I read about the issue with the duplicate check requests but that should not prevent from implementing the option, since even if it’s only for testing an pre-prod, also those machines might be / are managed by config management (puppet in this case)
- even tho I am not a fan of making things backwards breaking, I see your point and can adjust that to remove Sensu < 0.19 support

**Comment by RiRa12621** [comment]
is there nay progress to be expected on this one ?
it’s been a month now

**Comment by raffraffraff** [comment]
A lot more, if you include the length of time they ignored my pull
request... I am continuing to use my own fork. Good luck.

On 24 May 2017 14:18, "Rick Rackow" <notifications@github.com> wrote:

is there nay progress to be expected on this one ?
it’s been a month now

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/sensu/sensu-puppet/pull/639#issuecomment-303720661>,
or mute
the thread

**Comment by RiRa12621** [comment]
true actually

**Comment by jaxxstorm** [comment]
I've merged this for you.

I want to make it clear _why_ this took so long.

- This module is heavily relied on for people's monitoring systems. We have a responsibility to merge good code, and review it thoroughly. This takes time.
- This is a part time gig for me, I have a job that I need to do as well as maintain multiple open source projects

I understand the frustration here, but please try and be patience.

**Comment by shajithravi** [comment]
@jaxxstorm any updates on this?

**Pull request merged**